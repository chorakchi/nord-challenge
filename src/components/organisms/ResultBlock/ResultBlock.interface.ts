import { HTMLAttributes } from "react";

type HeaderItem = {
  key: string;
  label: string;
};

export interface TypographyProps extends HTMLAttributes<HTMLElement> {
  header?: Array<HeaderItem>;
  onClickHeader?: (key: string) => void;
  items?: Array<any>;
}
