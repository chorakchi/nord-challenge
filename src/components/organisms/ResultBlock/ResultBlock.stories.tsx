import { ComponentStory, ComponentMeta } from "@storybook/react";

import { ResultBlock } from "./ResultBlock";
import { mockData } from "./ResultBlock.mock";

export default {
  title: "organisms/ResultBlock",
  component: ResultBlock,
  parameters: {
    docs: {
      description: {
        component: "",
      },
    },
    backgrounds: {
      default: "light",
    },
  },
} as ComponentMeta<typeof ResultBlock>;

const Template: ComponentStory<typeof ResultBlock> = (args) => (
  <ResultBlock {...args} />
);

export const ResultBlock_ = Template.bind({});
ResultBlock_.args = {
  ...mockData,
};
