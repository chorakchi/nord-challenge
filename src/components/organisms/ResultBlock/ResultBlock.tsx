import { TypographyProps } from "./ResultBlock.interface";
import { Table } from "components";
export const ResultBlock: React.FC<TypographyProps> = ({
  header,
  items = [],
  onClickHeader,
  ...props
}: TypographyProps) => {
  return items.length ? (
    <div
      {...props}
      className="shadow-lg h-2/3 w-3/4 overflow-auto border-b border-gray-200 sm:rounded-lg"
    >
      <Table header={header} items={items} onClickHeader={onClickHeader} />
    </div>
  ) : (
    <></>
  );
};
