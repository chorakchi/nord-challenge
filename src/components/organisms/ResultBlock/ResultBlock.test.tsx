import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { ResultBlock } from "./ResultBlock";

describe("ResultBlock", () => {
  it("should render the ResultBlock", () => {
    const subject = shallow(<ResultBlock/>);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<ResultBlock/>).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
