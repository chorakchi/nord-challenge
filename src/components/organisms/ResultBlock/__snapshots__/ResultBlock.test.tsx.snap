// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultBlock should render correctly and match to the Snapshot 1`] = `null`;
